### ISSUE TITLE
Point_set_3 remove(const Index&) error

### ISSUE THREAD

**Issue by unclejimbo** [CGAL/cgal#2221]
## Issue Details

Using the `remove(const Index&)` method in Point_set_3 package generate the following error:

> error C2664: 'void std::swap(std::exception_ptr &,std::exception_ptr &) throw()': cannot convert argument 1 from 'const CGAL::Point_set_3<Point_3,CGAL::Vector_3<Kernel_>>::Index' to 'std::exception_ptr &'

## Source Code

```cpp
#include <CGAL/Simple_cartesian.h>
#include <CGAL/Point_set_3.h>

using Kernel = CGAL::Simple_cartesian<float>;
using Point_3 = Kernel::Point_3;
using Point_set = CGAL::Point_set_3<Point_3>;

int main()
{
	Point_set ps;
	for (auto i = 0; i < 10; ++i) {
		ps.insert();
	}

	size_t i = 0;
	for (auto iter = ps.begin(); iter != ps.end(); ++iter) {
		if (i++ % 3 == 0) {
			ps.remove(*iter);
		}
	}
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Win10 x64
* Compiler: vs2017
* Release or debug mode: debug & release
* Specific flags used (if any):
* CGAL version: 4.10
* Boost version: 1.64
* Other libraries versions if used (Eigen, TBB, etc.): Eigen 3.3.4

**Comment by lrineau** [comment]
I can reproduce the bug on my Linux machine, and the issue is very simple. See the code of `Point_set_3::remove():`
```c++
  void remove (const Index& index)
  {
    std::swap (index, *(end() - 1));
    ++ m_nb_removed;
  }
```

As `index` is `const`, it cannot be swapped.

**Comment by lrineau** [comment]
That function is not tested. Only its `remove(iterator)` counterpart is tested.

**Comment by lrineau** [comment]
That function could be replaced by:
```C++
  void remove (const Index& index)
  {
    remove(m_indices.begin() + index);
  }
```

@unclejimbo Could you please try that patch?

```patch
diff --git a/Point_set_3/include/CGAL/Point_set_3.h b/Point_set_3/include/CGAL/Point_set_3.h
index b18e939..058ae65 100644
--- a/Point_set_3/include/CGAL/Point_set_3.h
+++ b/Point_set_3/include/CGAL/Point_set_3.h
@@ -581,8 +581,7 @@ public:
   */
   void remove (const Index& index)
   {
-    std::swap (index, *(end() - 1));
-    ++ m_nb_removed;
+    remove(m_indices.begin() + index);
   }
 
 
```

**Comment by unclejimbo** [comment]
@lrineau You are right, this one works.

**Comment by sgiraudot** [comment]
Not related to this bug but from what I see in your code: [the documentation of Point_set_3] states that `remove()` invalidates all iterators, pointers and references, so iterating on the point set while removing items from it may lead to unexpected behaviors or errors.

**Comment by unclejimbo** [comment]
@sgiraudot Thanks, I know that, this is only for demonstration.

**Comment by lrineau** [comment]
This will be fixed in the future CGAL version 4.10.1, now that the PR #2230 is merged.
